editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/integration/properties/creators/NumberParameterPropertyUICreatorIT.java

```diff
@@ -74,4 +80,57 @@ public void testUIDisplayedForAdvanced() throws Exception {
 		assertThat(PropertiesUtils.getTypedPropertyFromUri(camelModelElement, parameter, PropertiesUtils.getComponentFor(camelModelElement))).isEqualTo("2");
 	}
 
+	@Test
+	public void testValidDurationConversion() throws Exception {
+		ICamelManagerService svc = CamelServiceManagerUtil.getManagerService(CamelModelFactory.getLatestCamelVersion());
+		assertNotNull("Camel Catalog Service was null!", svc);
+
+		String duration = "1h10m30s";
+		long millis = svc.durationToMillis(duration);
+		assertEquals(1*60*60*1000 + 10 * 60 * 1000 + 30 * 1000, millis);
+		
+		duration = "1h10m";
+		millis = svc.durationToMillis(duration);
+		assertEquals(1*60*60*1000 + 10 * 60 * 1000, millis);

```

Comment:
It allows to detect if a single check is broken or if everything is broken.
if only values are changing, we can also use Parameterized tests